recipes/pdal/all/conanfile.py

```diff
@@ -0,0 +1,105 @@
+import os
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+
+class PdalConan(ConanFile):
+    name = "pdal"
+    description = "PDAL is Point Data Abstraction Library. GDAL for point cloud data."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://pdal.io"
+    license = "BSD"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "with_xml": [True, False],
+               "with_zstd": [True, False],
+               "with_laszip": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "with_xml": True,
+                       "with_zstd": True,
+                       "with_laszip": True}
+    topics = ("conan", "pdal", "gdal")
+
+    _source_subfolder = "source_subfolder"
+
+    _cmake = None
+
+    def requirements(self):
+        self.requires("gdal/3.1.4")
+        self.requires("libgeotiff/1.6.0")
+        if self.options.with_xml:
+            self.requires("libxml2/2.9.10")
+        if self.options.with_zstd:
+            self.requires("zstd/1.4.5")
+        if self.options.with_laszip:
+            self.requires("laszip/3.4.3")
```

Comment:
I see more libraries here: https://github.com/PDAL/PDAL/blob/c5705896d1d45e51fc4a5a9fd5ee1a86dce9ea97/CMakeLists.txt#L265

eigen, boost, nlohmann_json in include dirs.

libgeotiff in libraries (comes from gdal or is it also a direct dependency of pdal?), and also lzma.

Vendoring? Optional dependencies?

Reply:
New issue(s) ?

Reply:
add a TODO at least, because this kind of things tends to be forgotten + it will require a lot of work for next maintenair.

Reply:
Carefully checked deps and vendoring.

Deps not used at all:

- zlib is a dependency of a disabled plugin i3s

- openscenegraph, hdf, postgres, tiledb, nitro, fbx, mrsid - dependencies of a plugins openscenegraph, hdf, pgpointcloud, tiledb, nitf, fbx, mrsid

- jni, jace, geowave - disabled plugin geowave

- gtest: we do not use testing, disabled

- kazhdan: not a standalone library I think


Should work on:

- libunwind - it is used if found, should switch to CCI explicitly

- nanoflann, nlohmann, pdalboost: yes, we should carefully switch from vendoring to CCI after identifying versions and build options

- arbiter - should be switched to CCI when package would be available

- openssl, curl - required only in arbiter, now they are disabled


I have added a TODOs. I think they deserve own tickets.

Reply:
While waiting for gdal packages, I have addressed some of those improvements.

1. Added libunwind (tested for Linux)

2. Switched to nanoflann from CCI

3. Using nlohmann_json is difficult. pdal vendored version 3.6.1 with [several changes](https://gist.github.com/theirix/6c6f768b36b078380d49fba50be98b64). Added alias `namespace NL = nlohmann;` requires us to replace every reference `NL::` or to place alias before `#include` because there is no central place where `json.hpp` is included.